### DISCUSSION TITLE
Allow packages to opt out of package compatibility reporting

### DISCUSSION THREAD

**Discussion by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#2938] [Ideas]
This has come up a few times in the past, but most recently here: https://github.com/SwiftPackageIndex/PackageList/pull/6587#issuecomment-1994451498

Some packages are not going to compile in a reasonable amount of time for a build system like ours where we need to monitor and build thousands of packages. Leaving the build matrix looking like nothing works, when the package may work, it's just that we couldn't compile it in time.

We could allow a setting in the `.spi.yml` manifest for package authors to indicate we shouldn't build their package. This would save us time in the build system not having to attempt builds we know will fail after a significant amount of time, and remove a misleading piece of information from the package page.

The flag should both inhibit triggering of builds for the package and also the display of the whole "Package Compatibility" section on the package page. We could maybe include a message saying the package has opted out, but I'm not sure if that's a good idea yet. Needs more thought.

cc @furby-tm

**Comment by Sherlouk** [comment]
As a consumer, this would be a huge blow for me personally. Admittedly it's a tiny fraction of packages (that need to be opted out due to timeouts, unclear who would otherwise opt themselves out) - but in those cases would be very unfortunate.

**Reply by heckj** [reply]
I'm right there with you James - while not 100% absolutely critical, it's part of the value-add that I love about this site and service.
